pkg/kfake/01_fetch.go

```diff
@@ -129,6 +129,11 @@ func (c *Cluster) handleFetch(creq *clientReq, w *watchFetch) (kmsg.Response, er
 		sp := kmsg.NewFetchResponseTopicPartition()
 		sp.Partition = p
 		sp.ErrorCode = errCode
+		// Encode empty record sets as an empty-but-present byte slice rather
+		// than null. A nil slice would be serialized as a length of -1 (or 0
+		// in the compact-nullable form), which some clients (for example,
+		// librdkafka) interpret as an invalid MessageSet size.
+		sp.RecordBatches = make([]byte, 0)
```

Comment:
I'm okay to add this here, but I'm fairly sure this if running librdkafka is failing on this, it's technically a librdkafka bug:

https://github.com/apache/kafka/blob/d04f171d3123a158a32ea65eb59600bc2f0fd628/generator/src/main/java/org/apache/kafka/message/FieldType.java#L292-L295

I do remember being surprised when implementing the serializing years ago that, technically, `records` could be nullable.